core/src/main/java/com/linecorp/armeria/common/metric/ServiceNaming.java

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright 2021 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+package com.linecorp.armeria.common.metric;
+
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.common.logging.RequestOnlyLog;
+import com.linecorp.armeria.common.util.Unwrappable;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceConfig;
+
+/**
+ * Generates the default name of a {@link com.linecorp.armeria.server.Service} from its {@link ServiceConfig}.
```

Comment:
nit: (A global comment) You don't need to use an FQCN for Javadoc. `{@link Service}` is enough.